lib/transforms/type_legalizer.cc

```diff
@@ -1232,6 +1232,15 @@ static void RunOnInstruction(TFIDFVectorizeInst* inst) {
   }
 }
 
+static void RunOnInstruction(SelectInst* inst) {
+  const auto& ty0 = inst->GetOperand(0).GetType();
+  const auto& elem_ty = inst->GetOperand(1).GetType().GetDataType();
+  if (!ty0.IsValid() || elem_ty == DataType::INVALID) {
+    return;
+  }
+  inst->GetResultsTypes()[0] = Type{elem_ty, ty0.GetDimSizes()};
```

Comment:
The result type is tricky if broadcasting is allowed. And TF(1.X) and ONNX have different restrictions. Can we set the result type to input1's type instead of the condition input's?

Reply:
right. For ONNX, the broadcasting is based on Op0 (condition) but for TF, broadcasting is aligned with X/Y.

Let's choose the largest shape?